uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -0,0 +1,485 @@
+      });
+
+      return leaveStatus ? leaveStatus.value : false;
+    }
+
+    /**
+     * Indexes for easy access the data that the component needs
+     */
+    function indexData () {
+      vm.supportData.dayTypes = _.indexBy(vm.supportData.dayTypes, 'name');
+      vm.supportData.leaveRequestStatuses = _.indexBy(vm.supportData.leaveRequestStatuses, 'value');
+      vm.supportData.publicHolidays = _.transform(vm.supportData.publicHolidays, function (result, publicHoliday) {
+        result[dateObjectWithFormat(publicHoliday.date).valueOf()] = publicHoliday;
+      }, {});
+    }
+
+    /**
+     * Index leave requests by contact_id as first level
+     * and date as second level
+     *
+     * @param  {Array} leaveRequestsList
+     * @return {Promise}
+     */
+    function indexLeaveRequests (leaveRequestsList) {
+      var deferred = $q.defer();
+
+      leaveRequestsList.forEach(function (leaveRequest) {
+        var days = leaveRequestDays(leaveRequest);
+
+        leaveRequests[leaveRequest.contact_id] = leaveRequests[leaveRequest.contact_id] || {};
+
+        days.forEach(function (day) {
+          leaveRequests[leaveRequest.contact_id][day.date] = leaveRequest;
+        });
+      });
+
+      deferred.resolve();
+
+      return deferred.promise;
+    }
+
+    /**
+     * Initializes the event listeners
+     */
+    function initListeners () {
+      $rootScope.$on('LeaveCalendar::showMonths', showMonthIfInList);
+      $rootScope.$on('LeaveRequest::new', addLeaveRequest);
+      $rootScope.$on('LeaveRequest::edit', updateLeaveRequest);
+      $rootScope.$on('LeaveRequest::updatedByManager', updateLeaveRequest);
+      $rootScope.$on('LeaveRequest::deleted', deleteLeaveRequest);
+    }
+
+    /**
+     * Returns whether a date is of a specific type
+     * half_day_am or half_day_pm
+     *
+     * @param  {string} name
+     * @param  {object} leaveRequest
+     * @param  {string} date
+     *
+     * @return {boolean}
+     */
+    function isDayType (name, leaveRequest, date) {
+      var dayType = vm.supportData.dayTypes[name];
+
+      if (moment(date).isSame(leaveRequest.from_date)) {
+        return dayType.value === leaveRequest.from_date_type;
+      }
+
+      if (moment(date).isSame(leaveRequest.to_date)) {
+        return dayType.value === leaveRequest.to_date_type;
+      }
+    }
+
+    /**
+     * Returns whether a leaveRequest is pending approval
+     *
+     * @param  {object} leaveRequest
+     * @return {boolean}
+     */
+    function isPendingApproval (leaveRequest) {
+      var status = vm.supportData.leaveRequestStatuses[leaveRequest.status_id];
+
+      return status.name === sharedSettings.statusNames.awaitingApproval;
+    }
+
+    /**
+     * Decides whether sent date is a public holiday
+     *
+     * @param  {string} date
+     * @return {boolean}
+     */
+    function isPublicHoliday (date) {
+      return !!vm.supportData.publicHolidays[dateObjectWithFormat(date).valueOf()];
```

Comment:
Not the same thing, `_.includes` is to finding items in an array, here `vm.supportData.publicHolidays` is an object